openmc/geometry.py

```diff
@@ -455,3 +455,105 @@ def get_lattices_by_name(self, name, case_sensitive=False, matching=False):
         lattices = list(lattices)
         lattices.sort(key=lambda x: x.id)
         return lattices
+
+    def count_cell_instances(self, universe=None):
```

Comment:
Seems like maybe this should be a method of `Universe`. `Geometry` could have its own version that resets the instance counts and then calls `self.root_universe.count_cell_instances()`. What do you think?